datadog/api/metrics.py

```diff
@@ -42,8 +43,19 @@ def rec_parse(points_lst):
                     return []
 
                 point = points_lst.pop()
-                timestamp = now if isinstance(point, Number) else point[0]
-                value = float(point) if isinstance(point, Number) else float(point[1])
+                if isinstance(point, Number):
+                    timestamp = now
+                    value = float(point)
+                # Distributions contain a list of points
+                else:
+                    timestamp = point[0]
+                    if isinstance(point[1], list):
+                        float_points = []
+                        for p in point[1]:
+                            float_points.append(float(p))
+                        value = float_points
+                    else:
+                        value = float(point[1])
```

Comment:
Better to use list comprehensions here and relax type requirements a bit:

```python
from collections import Iterable

timestamp = point[0]
# more flexible - allows clients to use iterables other than the built-in list
if isinstance(point[1], Iterable):
    # more terse and faster
    value = [float(p) for p in point[1]]
else:
    value = float(point[1])
```

Would like to see some tests as well.

Also, this is outside of the scope of this PR but we should convert `rec_parse` to a non-recursive function for simplicity, speed and memory savings.